src/utils/getFilteredTodos.ts

```diff
@@ -0,0 +1,27 @@
+import { StatusFilter } from '../modules/StatusFilter';
+import { Todo } from '../types/Todo';
+
+export function getFilteredTodos(
+  todos: Todo[],
+  filters: { query: string; status: StatusFilter },
+) {
+  let filteredTodos = [...todos];
+
+  if (filters.status !== StatusFilter.ALL) {
+    if (filters.status === StatusFilter.ACTIVE) {
+      filteredTodos = filteredTodos.filter(todo => !todo.completed);
+    } else {
+      filteredTodos = filteredTodos.filter(todo => todo.completed);
+    }
+  }
+
+  const normalizedQuery = filters.query.toLowerCase();
+
+  if (normalizedQuery) {
+    filteredTodos = filteredTodos.filter(todo =>
+      todo.title.toLowerCase().includes(normalizedQuery),
+    );
+  }
+
+  return filteredTodos;
```

Comment:
This function can be optimized by combining the two filters into one. Currently, you're making two passes over the `todos` array, one for the status filter and one for the query filter. You can combine these into a single `filter` call which checks both conditions. This will make your function faster for large `todos` arrays.